Makefile

```
@@ -33,6 +33,7 @@ $(API_SPEC_PATH):
 	rm -rf $(API_CLONE_PATH)
 
 .PHONY: api-generate
+api-generate: old_spec := $(shell ls -1 $(API_SERVICE_PATH)/openapi.*.yaml)
 api-generate: $(OPENAPI_GENERATOR) clean-openapi $(API_SPEC_PATH)
```

I'd suggest this:

1. Remove `.PHONY api-generate`
2.
```suggestion
api-generate: $(API_CLIENT_CODE_PATH)/index.js
$(API_CLIENT_CODE_PATH)/index.js: $(OPENAPI_GENERATOR) clean-openapi $(API_SPEC_PATH)
```

This is more idiomatic: if the file does not exist or is older than the spec or the spec name has changed, it gets regenerated.

---

I like `2`, but I do not understand the purpose of `1`.
Why removing the `.PHONY api-generate` if `api-generate` is not a regular file to be generated by `make api-generate`?